tests/test_wasm_builtin_simd.c

```diff
@@ -0,0 +1,862 @@
+  expect_vec(f32x4_replace_lane_last((f32x4){0, 0, 0, 0}, 53), ((f32x4){0, 0, 0, 53}));
+
+  // f64x2 lane accesses
+  expect_vec(f64x2_splat(-5), ((f64x2){-5, -5}));
+  expect_eq(f64x2_extract_lane_first((f64x2){-5, 0}), -5);
+  expect_eq(f64x2_extract_lane_last((f64x2){0, -5}), -5);
+  expect_vec(f64x2_replace_lane_first((f64x2){0, 0}, 53), ((f64x2){53, 0}));
+  expect_vec(f64x2_replace_lane_last((f64x2){0, 0}, 53), ((f64x2){0, 53}));
+
+  // i8x16 comparisons
+  expect_vec(
+    i8x16_eq(
+      (i8x16){0, 127, 13, 128,  1,  13, 129,  42, 0, 127, 255, 42,   1,  13, 129,  42},
+      (i8x16){0, 255, 13, 42, 129, 127,   0, 128, 0, 255,  13, 42, 129, 127,   0, 128}
+    ),
+    ((i8x16){-1, 0, -1, 0, 0, 0, 0, 0, -1, 0, 0, -1, 0, 0, 0, 0})
+  );
+  expect_vec(
+    i8x16_ne(
+      (i8x16){0, 127, 13, 128,  1,  13, 129,  42, 0, 127, 255, 42,   1,  13, 129,  42},
+      (i8x16){0, 255, 13, 42, 129, 127,   0, 128, 0, 255,  13, 42, 129, 127,   0, 128}
+    ),
+    ((i8x16){0, -1, 0, -1, -1, -1, -1, -1, 0, -1, -1, 0, -1, -1, -1, -1})
+  );
+  expect_vec(
+    i8x16_lt_s(
+      (i8x16){0, 127, 13, 128,  1,  13, 129,  42, 0, 127, 255, 42,   1,  13, 129,  42},
+      (i8x16){0, 255, 13, 42, 129, 127,   0, 128, 0, 255,  13, 42, 129, 127,   0, 128}
+    ),
+    ((i8x16){0, 0, 0, -1, 0, -1, -1, 0, 0, 0, -1, 0, 0, -1, -1, 0})
+  );
+  expect_vec(
+    i8x16_lt_u(
+      (i8x16){0, 127, 13, 128,  1,  13, 129,  42, 0, 127, 255, 42,   1,  13, 129,  42},
+      (i8x16){0, 255, 13, 42, 129, 127,   0, 128, 0, 255,  13, 42, 129, 127,   0, 128}
+    ),
+    ((i8x16){0, -1, 0, 0, -1, -1, 0, -1, 0, -1, 0, 0, -1, -1, 0, -1})
+  );
+  expect_vec(
+    i8x16_gt_s(
+      (i8x16){0, 127, 13, 128,  1,  13, 129,  42, 0, 127, 255, 42,   1,  13, 129,  42},
+      (i8x16){0, 255, 13, 42, 129, 127,   0, 128, 0, 255,  13, 42, 129, 127,   0, 128}
+    ),
+    ((i8x16){0, -1, 0, 0, -1, 0, 0, -1, 0, -1, 0, 0, -1, 0, 0, -1})
+  );
+  expect_vec(
+    i8x16_gt_u(
+      (i8x16){0, 127, 13, 128,  1,  13, 129,  42, 0, 127, 255, 42,   1,  13, 129,  42},
+      (i8x16){0, 255, 13, 42, 129, 127,   0, 128, 0, 255,  13, 42, 129, 127,   0, 128}
+    ),
+    ((i8x16){0, 0, 0, -1, 0, 0, -1, 0, 0, 0, -1, 0, 0, 0, -1, 0})
+  );
+  expect_vec(
+    i8x16_le_s(
+      (i8x16){0, 127, 13, 128,  1,  13, 129,  42, 0, 127, 255, 42,   1,  13, 129,  42},
+      (i8x16){0, 255, 13, 42, 129, 127,   0, 128, 0, 255,  13, 42, 129, 127,   0, 128}
+    ),
+    ((i8x16){-1, 0, -1, -1, 0, -1, -1, 0, -1, 0, -1, -1, 0, -1, -1, 0})
+  );
+
+  // expected {0x00ffffff, 0xff00ffff, 0xff00ffff, 0xff00ffff},
+  //      got {0x0000ffff, 0xffffffff, 0x0000ffff, 0xffffffff}
+  expect_vec(
+    i8x16_le_u(
+      (i8x16){0, 127, 13, 128,  1,  13, 129,  42, 0, 127, 255, 42,   1,  13, 129,  42},
+      (i8x16){0, 255, 13, 42, 129, 127,   0, 128, 0, 255,  13, 42, 129, 127,   0, 128}
+    ),
+    ((i8x16){-1, -1, -1, 0, -1, -1, 0, -1, -1, -1, 0, -1, -1, -1, 0, -1})
+  );
```

Comment:
V8 is giving incorrect results here.

Reply:
i8x16.le_u incorrectly returns 0 in lane comparing 13 with 13, -1 in lane comparing 129 with 0, and 0 in lane comparing 42 with 128.

Reply:
I find it strange that the lane comparing 0 with 0 is fine but the lane comparing 13 with 13 is wrong :/

Reply:
Took a quick look, and we already comparison tests that test these values, so this is weird to me too. Could you file a V8/Chromium for tracking with a repro?

Reply:
https://bugs.chromium.org/p/v8/issues/detail?id=8635